### DISCUSSION TITLE
Qraukus, hashicorp vault  , multiple mount path in the properties quarkus.vault.kv-secret-engine-mount-path

### DISCUSSION THREAD

**Discussion by chakerfezai** [quarkusio/quarkus#36061] [Q&A]
![image](https://github.com/quarkusio/quarkus/assets/29119181/942e0abf-5ea8-4c3c-a658-0d22314eec5c)

[a comment by an automated account is not shown]

**Comment by anunnakian** [comment]
You can create a class and use the same source code as the method VaultKvManager.readSecret(String) located on quarkus-vault extension.

Here is an example using quarkus-vault 1.1.0 :

```java
public Map<String, String> readSecret(String mount, String path) {
  String clientToken = this.vaultAuthManager.getClientToken();
  VaultKvSecretV2 secret = this.vaultInternalKvV2SecretEngine.getSecret(clientToken, mount, path);
  return secret.data.data;
}
```

Your can inject all dependencies by using @Inject annotation.

**Reply by chakerfezai** [reply]
Hello Mohammed,

thank you for your answer.

Thank you for your answer, indeed we have opted for this solution.

It would be interesting to have this possibility to do so at the level of the properties file.

Thanks.
Chaker.
